src/security/cert-storage.ts

```diff
@@ -135,17 +136,92 @@ export class CertStorage implements SecurityAgent {
     storage: Storage,
     fw: Forwarder,
     prvKeyBits: Uint8Array,
-    interestLifetime = 5000,
   ) {
     const result = new CertStorage(
       trustAnchor,
       ownCertificate,
       storage,
       fw,
       prvKeyBits,
-      interestLifetime,
     );
     await result.readyEvent;
     return result;
   }
+
+  /**
+   * Modified version of getCertificate function, designed for trying to
+   * fetch PoR from all other already trusted parties
+   * @param strangerKeyname First part of PoR name, i.e. "/<𝑍𝑏 name>/KEY/<𝑇𝑏 keyid>"
+   */
+  async fetchPoR(strangerKeyname: string) {
+    // loop through trusted certificates
+    for (const item of this.trustedNames) {
+      // Construct <Z_a name encoded>
+      const parts: string[] = item.split('/');
+      let partsBeforeKey: string = '';
+      for (let i = 0; i < parts.length; i++) { //put <Z_a name encoded> together with a loop
+        if (parts[i].includes('KEY')) {
+          break;
+        }
+        partsBeforeKey += parts[i] + '/';
+      }
+
+      // return if we found that one of the users we trust, also trusts the new user
+      const response: Certificate | undefined = await this.getCertificate(
+        new Name(strangerKeyname + partsBeforeKey + 'v1'),
+        false,
+      );
+      if (response != undefined) {
+        return response;
+      }
+
+      // If not getting at least one POR, return undefined
+      return undefined;
+    }
+  }
+
+  async generatePoR(
+    strangerKeyname: string, // name of t_b, i.e. "/<𝑍𝑏 name>/KEY/<𝑇𝑏 keyid>"
+    certName: Name, // follow PoR naming format
+  ): Promise<Certificate> {
+    // Request POR certificate issued by every other known (authenticated) user
+    const foreignPoR: Certificate | undefined = await this.fetchPoR(strangerKeyname);
+
+    // If not getting at least one POR, abort (treat as data verification failure)
+    if (foreignPoR == undefined) {
+      throw new Error('Data verification failure: No POR certificate found issued by trusted user');
+    }
```

Comment:
This reads as "a PoR cannot be generated without an existing one".
Then, how can the first PoR be generated?